ballerina/tests/secure_ftps_client_endpoint_test.bal

```diff
@@ -0,0 +1,514 @@
+        secureSocket: {
+            key: {
+                path: "tests/resources/keystore.jks",
+                password: "changeit"
+            },
+            cert: {
+                path: "tests/resources/keystore.jks",
+                password: "changeit"
+            },
+            mode: IMPLICIT,
+            dataChannelProtection: PRIVATE
+        }
+    }
+};
+
+// Create the config with CLEAR data channel protection for testing
+ClientConfiguration ftpsClearDataChannelConfig = {
+    protocol: FTPS,
+    host: "127.0.0.1",
+    port: 21214,
+    auth: {
+        credentials: {username: "wso2", password: "wso2123"},
+        secureSocket: {
+            key: {
+                path: "tests/resources/keystore.jks",
+                password: "changeit"
+            },
+            cert: {
+                path: "tests/resources/keystore.jks",
+                password: "changeit"
+            },
+            mode: EXPLICIT,
+            dataChannelProtection: CLEAR
+        }
+    }
+};
+
+// Config to test default port logic. 
+// We set port 21, and the backend should detect IMPLICIT mode and swap it to 990.
+ClientConfiguration ftpsImplicitDefaultPortConfig = {
+    protocol: FTPS,
+    host: "127.0.0.1",
+    port: 21, 
+    auth: {
+        credentials: {username: "wso2", password: "wso2123"},
+        secureSocket: {
+            key: {path: "tests/resources/keystore.jks", password: "changeit"},
+            cert: {path: "tests/resources/keystore.jks", password: "changeit"},
+            mode: IMPLICIT
+        }
+    }
+};
+
+Client? ftpsExplicitClientEp = ();
+Client? ftpsImplicitClientEp = ();
+Client? ftpsClearDataChannelClientEp = ();
+
+// Root path for client isolation
+const string FTPS_CLIENT_ROOT = "/ftps-client";
+
+@test:BeforeSuite
+function initFtpsTestEnvironment() returns error? {
+    io:println("Initializing FTPS test clients");
+    ftpsExplicitClientEp = check new (ftpsExplicitConfig);
+    ftpsImplicitClientEp = check new (ftpsImplicitConfig);
+    ftpsClearDataChannelClientEp = check new (ftpsClearDataChannelConfig);
+    
+    // Clean the sandbox
+    check cleanFtpsTarget();
+}
+
+@test:AfterSuite
+function cleanupFtpsTestEnvironment() returns error? {
+    io:println("Cleaning up FTPS test files...");
+    check cleanFtpsTarget();
+}
+
+function cleanFtpsTarget() returns error? {
+    if ftpsExplicitClientEp is () {
+        return;
+    }
+    // Helper to clean up specific files used in tests if they exist
+    string[] files = ["file2.txt", "tempFtpsFile1.txt", "tempFtpsFile2.txt", "tempFtpsPrivate.txt", 
+    "tempFtpsClear.txt", "tempFtpsFile3.txt", "tempFtpsFile4.txt", "tempFtpsFile5.txt", "tempFtpsFile6.txt"];
+    foreach string f in files {
+        var deleteResult = trap (<Client>ftpsExplicitClientEp)->delete(FTPS_CLIENT_ROOT + "/" + f);
+        // Ignore errors for files that don't exist
+        if deleteResult is error {
+            // Silently ignore - file may not exist
```

Comment:
Let's remove this part. We can put the files beforehand to make sure the file delete works properly.